### PR TITLE
React2 week1

### PR DESCRIPTION
**Pull request by Mabonye**
Meal sharing React2-week1 homework

### CODE REVIEW

File: src/backend/api/meals.js [hunk 1]
```diff
@@ -1,15 +1,229 @@
...
-router.get("/", async (request, response) => {
-  try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
-  } catch (error) {
-    throw error;
-  }
+// GET 
+mealsRouter.get("/", async (req, res) => {
```
Comment (LucasIversen) [inline src/backend/api/meals.js:6]: Looks good 👌

File: src/backend/api/meals.js [hunk 2]
```diff
@@ -1,15 +1,229 @@
...
+        const mealList = await meals;
+        res.json(mealList);
+    } catch (error) {
+        throw error;
+    }
+});
+
+
+// POST /
+mealsRouter.post("/", async (req, res) => {
```
Comment (LucasIversen) [inline src/backend/api/meals.js:114]: Good

File: src/backend/api/meals.js [hunk 3]
```diff
@@ -1,15 +1,229 @@
...
+    } catch (err) {
+        response.status(500).json({
+            message: "Internal Server Error"
+        });
+    }
+});
+
+
+
+module.exports = mealsRouter;
```
Comment (LucasIversen) [inline src/backend/api/meals.js:227]: Server seems good

File: src/backend/api/reservations.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    res.status(204).end();
+  } catch (error) {
+    res.status(500).json({
+      error: 'An error occurred while deleting a reservation'
+    });
+  }
+});
+
+
+module.exports = reservationsRouter;
```
Comment (LucasIversen) [inline src/backend/api/reservations.js:110]: good

File: src/backend/api/reviews.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+      response.status(204).end();
+   } catch (error) {
+      response.status(500).json({
+         error: 'An error occurred while deleting a review'
+      });
+   }
+})
+
+module.exports = reviewsRouter;
```
Comment (LucasIversen) [inline src/backend/api/reviews.js:102]: Goood 👍

File: src/client/App.js [hunk 6]
```diff
@@ -1,21 +1,15 @@
...
-        <p>lol</p>
-      </Route>
-      <Route exact path="/test-component">
-        <TestComponent></TestComponent>
-      </Route>
-    </Router>
-  );
+    return (
+        <Router>
+            <Route exact path="/">
```
Comment (LucasIversen) [inline src/client/App.js:8]: good routing

File: src/client/components/MealsList/Meal/Meal.jsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import "./Meal.css";
+
+function Meal(props) {
+    return (
+        <div key={props.id} className='meal'>
```
Comment (LucasIversen) [inline src/client/components/MealsList/Meal/Meal.jsx:7]: If you want to make it a list with list keys, make the outer container a <li> and wrap the map in a <ul>. it's up to use if you want to do that though

File: src/client/components/MealsList/MealsList.jsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import React, { useState, useEffect } from 'react';
+import "./MealsList.css";
+import Meal from './Meal/Meal';
+
+function MealsList() {
+    const [meals, setMeals] = useState([]);
+
+    useEffect(() => {
+        fetch('http://localhost:3005/all-meals')
```
Comment (LucasIversen) [inline src/client/components/MealsList/MealsList.jsx:9]: good fetch

File: src/client/components/MealsList/MealsList.jsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+            .then(response => response.json())
+            .then(data => setMeals(data))
+            .catch(error => console.error(error));
+    }, [meals]);
```
Comment (LucasIversen) [inline src/client/components/MealsList/MealsList.jsx]: Should the data be fetched every time the meals state change, or just on component mount []..?

**Review by LucasIversen — COMMENTED** [review]
Hi Jemima. Just a few comments from me.. Good job 👍

File: src/client/components/MealsList/Meal/Meal.jsx (hunk 7 above)
Comment (Mabonye) [reply src/client/components/MealsList/Meal/Meal.jsx:7]: Hello Lucas,
Thanks for the review, I have resolved the other comment but this one, i will do it in react2-week2.

**Pull request merged**